lib/m3u8-parser.js

```diff
@@ -8,9 +8,14 @@ const util     = require('util');
  * @extends WritableStream
  * @constructor
  */
-var m3u8parser = module.exports = function() {
+var m3u8parser = module.exports = function(ytdlOptions) {
   var lastLine = '';
   var self = this;
+  self.ytdlSavedLines = [];
```

Comment:
We must save the lines from the data bus, as we need them ALL if we're ever going to add backwards-time frames as an option.